packages/zend-log/library/Zend/Log/Writer/Stream.php

```diff
@@ -75,7 +75,17 @@ public function __construct($streamOrUrl, $mode = null)
                 $streamOrUrl = $streamOrUrl['stream'];
             }
 
-            if (! $this->_stream = @fopen($streamOrUrl, $mode, false)) {
+            try {
+                $this->_stream = @fopen($streamOrUrl, $mode, false);
```

Comment:
So, there's such test:
- https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/tests/Zend/Log/Writer/StreamTest.php#L49-L60

And this no longer returns a `resource` but a `XmlParser` class: `$resource = xml_parser_create()`

The test expected this code path to be executed:
- https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/packages/zend-log/library/Zend/Log/Writer/Stream.php#L61-L65

but instead it ends up here, with open throwing with TypeError:
- https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/packages/zend-log/library/Zend/Log/Writer/Stream.php#L78-L82

> `Failed asserting that 'fopen(): Argument #1 ($filename) must be of type string, XmlParser given' matches PCRE pattern "/not a stream/i".`

@falkenhawk: how to proceed here?

Reply:
to keep the most backward compatible way, would need to change code to reach also code path where `is_resource` is true, but I can't just add `instanceof XmlParser` there! :D

Reply:
The current assertions are something like this:

```
/Users/glen/.local/bin/php80 /private/var/folders/_x/06xnfg5j58s_qvckwbk31x8w0000gn/T/ide-phpunit.php --configuration /Users/glen/scm/php/zf1/zf1s/phpunit.xml.dist --filter Zend_Log_Writer_StreamTest --test-suffix StreamTest.php /Users/glen/scm/php/zf1/zf1s/tests/Zend/Log/Writer
Testing started at 20:28 ...
PHPUnit 3.7.38 by Sebastian Bergmann.

Configuration read from /Users/glen/scm/php/zf1/zf1s/phpunit.xml.dist

F...F.FF...

Time: 76 ms, Memory: 4.00Mb

There were 4 failures:

1) Zend_Log_Writer_StreamTest::testConstructorThrowsWhenResourceIsNotStream
Failed asserting that 'fopen(): Argument #1 ($filename) must be of type string, XmlParser given' matches PCRE pattern "/not a stream/i".

/Users/glen/scm/php/zf1/zf1s/tests/Zend/Log/Writer/StreamTest.php:57
/Users/glen/scm/php/zf1/zf1s/vendor/zf1s/phpunit/composer/bin/phpunit:63

2) Zend_Log_Writer_StreamTest::testConstructorThrowsWhenStreamCannotBeOpened
Failed asserting that 'Path cannot be empty' matches PCRE pattern "/cannot be opened/i".

/Users/glen/scm/php/zf1/zf1s/tests/Zend/Log/Writer/StreamTest.php:92
/Users/glen/scm/php/zf1/zf1s/vendor/zf1s/phpunit/composer/bin/phpunit:63

3) Zend_Log_Writer_StreamTest::testWriteThrowsWhenStreamWriteFails
Failed asserting that 'fwrite(): supplied resource is not a valid stream resource' matches PCRE pattern "/unable to write/i".

/Users/glen/scm/php/zf1/zf1s/tests/Zend/Log/Writer/StreamTest.php:122
/Users/glen/scm/php/zf1/zf1s/vendor/zf1s/phpunit/composer/bin/phpunit:63

4) Zend_Log_Writer_StreamTest::testShutdownClosesStreamResource
Failed asserting that 'fwrite(): supplied resource is not a valid stream resource' matches PCRE pattern "/unable to write/i".

/Users/glen/scm/php/zf1/zf1s/tests/Zend/Log/Writer/StreamTest.php:138
/Users/glen/scm/php/zf1/zf1s/vendor/zf1s/phpunit/composer/bin/phpunit:63

FAILURES!
Tests: 11, Assertions: 14, Failures: 4.

Process finished with exit code 1

```

It's difficult to see these in CI, as unexpected errors from https://github.com/zf1s/zf1/pull/85 shadow tests in https://github.com/zf1s/zf1/pull/88 and vice versa.

Reply:
ping @falkenhawk

Reply:
@glensc perhaps transform this else 
https://github.com/zf1s/zf1/blob/8e1040b4a5c147f8751994a597253d9c460b10cd/packages/zend-log/library/Zend/Log/Writer/Stream.php#L73
into e.g.
`else if (is_string($streamOrUrl) || is_array($streamOrUrl)) {`
and add additional else block at the end to catch the rest?
```
else {
  throw new Zend_Log_Exception('Resource is not a stream');
}
```
🤔